module/extension/FileWatcher/src/FileWatcher.cpp

```diff
@@ -38,7 +38,7 @@ namespace module::extension {
         std::lock_guard<std::mutex> lock(reactor.paths_mutex);
 
         // Work out what path we are watching
-        std::array<char, 512> pathbuff;
+        std::array<char, 512> pathbuff{};
         pathbuff.fill('\0');
```

Comment:
I feel like this should be 
```suggestion
        std::array<char, 512> pathbuff = {0};
```

Reply:
Based on [this link](https://stackoverflow.com/a/18304935), I think that using braces without a value inside them value initialises them with zeros.